scipy/optimize/cobyla.py

```diff
@@ -36,7 +36,8 @@ def wrapper(*args, **kwargs):
 
 @synchronized
 def fmin_cobyla(func, x0, cons, args=(), consargs=None, rhobeg=1.0,
-                rhoend=1e-4, maxfun=1000, disp=None, catol=2e-4):
+                rhoend=1e-4, maxfun=1000, disp=None, catol=2e-4,
+                callback=None):
```

Comment:
Being a public function, we should add this as a keyword only argument for easier forward compatibility.
```suggestion
                *, callback=None):
```